EIPS/eip-6404.md

```diff
@@ -396,10 +370,25 @@ class RlpTxType(IntEnum):
     SET_CODE = 0x04
     SET_CODE_MAGIC = 0x05
 
+def calculate_transaction_intrinsic_gas(tx: Transaction) -> uint:
+    tx_data = tx.payload.data()
+    gas_cost = TX_BASE_COST # FIXME: Should this be defined from another EIP?
+
+    if hasattr(tx_data, "authorization_list"):
+        for auth in tx_data.authorization_list:
+            gas_cost += get_signature_gas_cost(auth.signature, expected_algorithm=expected_signature_algorithm)
+
+    gas_cost += get_signature_gas_cost(tx.signature, expected_algorithm=expected_signature_algorithm)
+
+    return gas_cost
+
 def validate_transaction(tx: Transaction):
     tx_data = tx.payload.data()
 
+    expected_signature_algorithm = None
+
     if hasattr(tx_data, "type_"):
+        expected_signature_algorithm = SECP256K1_ALGORITHM
```

Comment:
yeah looks good. the ones with `type_` are those converted from RLP, those have to be SECP256K1 always.
those without `type_` are 6493 ones, those can support whatever is available.